### PR TITLE
CW Issue #1938: Improve handling of errors from cwctl commands

### PR DESCRIPTION
**Pull request by eharris369**
Fixes https://github.com/eclipse/codewind/issues/1938

Common up the code for checking the result of cwctl commands.

### CODE REVIEW

File: dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/CoreUtil.java [hunk 1]
```diff
@@ -152,11 +151,10 @@ public static IPath appendPathWithoutDupe(IPath start, String finish) {
 	 * In: [ "Here", "Is", "Some Input" ]
 	 * Out: "Here, Is, Some Input"
 	 */
-	public static String toCommaSeparatedString(Collection<String> collection) {
+	public static String formatString(String[] strArray, String separator) {
```
Comment (jopit) [inline dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/CoreUtil.java]: A simpler implementation for `formatString` is
```
	public static String formatString(String[] strArray, String separator) {
		return Arrays.stream(strArray).collect(Collectors.joining(separator));
	}
```

File: dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/cli/CLIUtil.java [hunk 2]
```diff
@@ -214,20 +208,14 @@ private static void checkErrorResult(String[] command, ProcessResult result) thr
 		try {
 			if (result.getOutput() != null && !result.getOutput().isEmpty()) {
 				JSONObject obj = new JSONObject(result.getOutput());
-				if (obj.has(ERROR_KEY)) {
```
Comment (jopit) [inline dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/cli/CLIUtil.java]: Shouldn't this handle the case that `obj.has(ERROR_KEY)` is true but `obj.has(ERROR_DESCRIPTION_KEY)` is false?  If it's ok to ignore that case, maybe there should be a comment explaining why it's ok.

File: dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/CoreUtil.java (hunk 1 above)
Comment (eharris369) [reply dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/CoreUtil.java]: Nice.  Done.

File: dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/cli/CLIUtil.java (hunk 2 above)
Comment (eharris369) [reply dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/cli/CLIUtil.java]: I added a comment and merged the 2 methods so it is easier to see what it is doing.  Until the output from cwctl is all normalized I did not want to depend solely on the existence of an "error" key which seems too generic.

**Pull request merged**